icu4c/source/i18n/datefmt.cpp

```diff
@@ -519,9 +519,15 @@ DateFormat::createInstanceForSkeleton(
 //----------------------------------------------------------------------
 
 DateFormat* U_EXPORT2
-DateFormat::create(EStyle timeStyle, EStyle dateStyle, const Locale& locale)
+DateFormat::create(EStyle timeStyle, EStyle dateStyle, const Locale& inLocale)
 {
     UErrorCode status = U_ZERO_ERROR;
+    Locale locale(inLocale);
+    // If the locale is bogus, fallback to use default locale first.
```

Comment:
It looks like create() is internal/private.
Wouldn't it be cleaner to add a UErrorCode parameter, and fail here with an illegal-argument-error?

Reply:
I found a better fix, PTAL

Reply:
I found the answer to my own question: The public createXyZInstance() functions all don't take UErrorCode either, so we couldn't propagate any error :-(

Looking at your new fix now.